velox/functions/sparksql/JsonObjectKeys.h

```diff
@@ -51,10 +51,27 @@ struct JsonObjectKeysFunction {
     }
 
     for (auto field : jsonObject) {
+      if (isFatal(field.error())) {
+        // On-Demand only fully validates the values used and the structure
+        // leading to it.
+        return false;
+      }
       out.add_item().copy_from(std::string_view(field.unescaped_key()));
     }
     return true;
   }
+
+ private:
+  // TODO: After upgrade simdjson, we could replace with simdjson function
+  // relevant simdjson function:
+  // https://github.com/simdjson/simdjson/blob/master/include/simdjson/error-inl.h#L10-L12
+  bool isFatal(simdjson::error_code error) noexcept {
+    // simdjson::INCOMPLETE_ARRAY_OR_OBJECT and simdjson::TAPE_ERROR
+    // indicate a fatal error and follow from the fact that
+    // the document is not valid JSON.
```

Comment:
Suggestion for brevity.

> // indicate a fatal error and follow from the fact that
    // the document is not valid JSON.

// indicate the document is not valid JSON.

Reply:
Thanks, updated